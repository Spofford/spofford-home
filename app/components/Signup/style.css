.wrapper {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  padding-bottom: 2rem;
  margin:0px auto;
}

.form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 2rem;
}

.form p {
  text-align:center;
}

.input, .half-input {
  padding: 1rem 1rem;
  border-radius: 3px;
  border: 1px solid #ccc;
  font-size: 1.1em;
  outline: none;
}

.inputGroup {
  display: flex;
  flex-flow: column nowrap;
  padding: 10px 0;
}

.signup header.fixed, .login header.fixed {
  position:relative !important;
}

label input {
  width:100%;
}

.form {
  background-color: #F2F2F2;
  border-radius:5px;
}

.signup h2, .login h2 {
  text-align:center;
  margin:20px 0px;
}

.signup .green, .login .green {
  margin:0px auto;
  display:table;
}

.signup .bottom-links, .login .bottom-links {
  margin-top:40px;
  width:100%;
  float:left;
  text-align:center;
}

.signup .bottom-links a, .login .bottom-links a {
  margin-bottom:5px;
  display:block;
}

.signup .error, .login .error {
  text-transform:uppercase;
  text-align:center;
  color:red;
  margin-top: 20px;
  font-family:'Montserrat', sans-serif;
}

@media (min-width: 700px) {
  .form {
    width:400px;
    margin:0px auto;
    display:table;
  }
}
