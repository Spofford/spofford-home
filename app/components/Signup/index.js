import React from "react"
import cssModules from "react-css-modules"
import { connect } from "react-redux"
import style from "./style.css"
import Actions from "../../redux/actions"
import { Redirect, Link } from 'react-router-dom'
import { default as Header } from "../Header"

export class Signup extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      redirect: false
    }

    this.submit = this.submit.bind(this)
  }

  submit() {
    const user = {
      email: document.getElementById("signup-email").value,
      password: document.getElementById("signup-password").value,
      password_confirmation: document.getElementById("signup-password-confirmation").value,
      first_name: document.getElementById("signup-first-name").value,
      last_name: document.getElementById("signup-last-name").value,
      role_id:1,
    }
    this.props.dispatch(Actions.userNew(user))
  }

  componentDidMount() {

  }

  componentDidUpdate(prevProps) {
    if (prevProps.user.id != this.props.user.id) {
      this.setState({
        redirect: true
      })
    }
  }

  render() {


    if (this.state.redirect) {
      return <Redirect to='/submissions'/>;
    }



    return (
      <div className="signup">
        <Header />
        <div className="body-container">
        <div className="copy-container">
        <div className="wrapper">
        <h2>Sign Up</h2>
          <div className="form">
            <div className="inputGroup">
              <label>Email
              <input
                className="input"
                type="text"
                autoComplete="username"
                id="signup-email" />
                </label>
            </div>
            <div className="inputGroup">
              <label>First Name
              <input
                className="input"
                type="text"
                autoComplete="first-name"
                id="signup-first-name" />
                </label>
            </div>
            <div className="inputGroup">
            <label>Last Name
              <input
                className="input"
                type="text"
                autoComplete="last-name"
                id="signup-last-name" />
              </label>
            </div>
            <div className="inputGroup">
            <label>Password
              <input
                className="input"
                type="password"
                autoComplete="new-password"
                id="signup-password" />
              </label>
            </div>
            <div className="inputGroup">
              <label>Confirm Password
              <input
                className="input"
                type="password"
                autoComplete="new-password"
                id="signup-password-confirmation" />
              </label>
            </div>
            <button className="green" onClick={this.submit}>Submit</button>
            <div className="bottom-links">
              <Link to="login">I have an account</Link>
            </div>
          </div>
        </div>
        </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  user: state.user,
  isAuthenticated: state.auth.isAuthenticated
})

export default connect(mapStateToProps)(cssModules(Signup, style))
