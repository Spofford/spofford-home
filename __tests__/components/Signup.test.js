import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Signup from '../../app/components/Signup';

const mockStore = configureStore();
const initialState = {
  user: {
    email: "amhasler@gmail.com",
    first_name: "Adam",
    last_name: "Hasler",
    role: 1,
    id: 20,
    charges: []
  },
  error: {
    message: "Wrong Credentials"
  }
}
const store = mockStore(initialState);

describe('<Signup />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<Signup store={store} />);
      //const component = wrapper.dive();

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
