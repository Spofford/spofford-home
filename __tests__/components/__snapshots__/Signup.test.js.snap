// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> render() renders the component 1`] = `
<Signup
  dispatch={[Function]}
  error="Wrong Credentials"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={
    Object {
      "charges": Array [],
      "email": "amhasler@gmail.com",
      "first_name": "Adam",
      "id": 20,
      "last_name": "Hasler",
      "role": 1,
    }
  }
/>
`;
